mcp/src/main/java/io/modelcontextprotocol/server/McpStatelessAsyncServer.java

```diff
@@ -375,11 +382,31 @@ private McpStatelessRequestHandler<CallToolResult> toolsCallRequestHandler() {
 				.findAny();
 
 			if (toolSpecification.isEmpty()) {
-				return Mono.error(new McpError("Tool not found: " + callToolRequest.name()));
+				return Mono.error(new McpError(new JSONRPCResponse.JSONRPCError(McpSchema.ErrorCodes.INVALID_PARAMS,
+						"Unknown tool: invalid_tool_name", "Tool not found: " + callToolRequest.name())));
 			}
+			else {
+				return toolSpecification.get().callHandler().apply(ctx, callToolRequest).onErrorResume(error -> {
+					logger.error("Error calling tool: {}", callToolRequest.name(), error);
+
+					// TODO: Should we handle the McpError+JsonRcpError specaially (e.g.
+					// propagate)
+					// or always return a CallToolResult with isError=true?
+					if (error instanceof McpError mcpError && mcpError.getJsonRpcError() != null) {
```

Comment:
I'm not sure this this assumption is correct? should we thread McpError + JsonRpcError content differently? 
If so we have to implement this logic for the McpAsyncServer's tool handler. Otherwise remove this condition here.
@chemicL any thoughts?